### DISCUSSION TITLE
I keep getting a red/bad GPS error, and "Hardware Health", with two different GPS's (M8Q-5883 & M8Q Mini), on two different flight controllers (F722-WPX & F765-WING)

### DISCUSSION THREAD

**Discussion by jackgrrd** [iNavFlight/inav#7311] [Q&A]
## Current Behavior
I cannot get either of my GPS's to work. Both are producing a red GPS, and a bad "Hardware Health" alert. The FC will restart after a few seconds of it going into this "bad" state. Both GPS's will pulse the blue "lock" LED, and the FC will remain normal, but as soon as everything is connected to iNav, the errors/red starts popping up.

## Steps to Reproduce
1. After connecting to iNav, if GPS is enabled through both ports and config, after 3-4 seconds, the GPS icon at the top right will go red, and "Hardware Health" will go from green to red.
2. I've followed the GPS wiring/setup instructions from Mateksys.com, and all cables are on the correct pins.
3. I actually ordered the F765-WING because I was having so much trouble with my F722-WPX, and as soon as I got done soldering the pins and wires of the F765, this brand new FC repeated the "Hardware Health" issue with both GPS's.
4.

## Expected behavior
After connecting to iNav, if GPS is enabled through both ports and config, after 3-4 seconds, the GPS icon at the top right will go red, and "Hardware Health" will go from green to red.

## Additional context
Here is the dump from the F765 w/ the M8Q-5883 -> https://pastebin.com/VexKij4V
Here is the second dump from the F765 w/ the M8Q Mini -> https://pastebin.com/H5b1sY1q

---

- FC Board name and vendor:
INAV/MATEKF765 3.0.1 Jul  4 2021 / 14:33:56 (753c4479)
# GCC-9.3.1 20200408 (release)

**Comment by DzikuVx** [comment]
This suggest that you connect it in a wrong way. Are you sure RX to TX and TX to RX? SDA to SDA and SCL to SCL and GPS is powered?

**Reply by DroneManStan** [reply]
I get the same exact behavior as jackgrrd with my Rekon 7 Pro F22 FC.
Before I made my transition over to INAV, I observed my Rekon 7 Pro BNF behavior while using Betaflight; which seem to have no issues, and my GPS had a lock on 12 Satellites while in my home near a window.  So I know my GPS seems to have the ability work correctly.

**Reply by STRAYK145** [reply]
The GPS in Ainav burned red, it turned out that it was better to look at the instructions for Matek. RX and TX are inverted. When I connected the RX wire to the TX jack and TX to RX, everything worked and the GPS turned blue.

**Comment by STRAYK145** [comment]
The GPS in Ainav burned red, it turned out that it was better to look at the instructions for Matek. RX and TX are inverted. When I connected the RX wire to the TX jack and TX to RX, everything worked and the GPS turned blue.
